pkg/controllers/addons_controller.go

```diff
@@ -113,7 +132,7 @@ func processAddonLayer(l layers.Layer) error { // nolint:gocyclo // ok
 // +kubebuilder:rbac:groups=kraan.io,resources=addons,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=kraan.io,resources=addons/status,verbs=get;update;patch
 func (r *AddonsLayerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	ctx := context.Background()
+	ctx := r.Context
```

Comment:
If reconciler.Context is set to context.Background in NewReconciler could this be simplified to ctx := context.Background()? Or does the context have to live on the struct? Convention in Go is that if a context is needed that it should be passed as the first argument to a function, so it could be created here and passed as an argument to child function calls.